### PR TITLE
3527 You can confirm outbound shipments with only placeholder lines

### LINKED ISSUE #3527: You can confirm outbound shipments with only placeholder lines
## What went wrong? 😲
 

Found During release testing: https://github.com/msupply-foundation/open-msupply/issues/3489#issuecomment-2034212191

Blank Internal order can be sent

Screenshot 2024-04-03 at 15 59 43

Video Reference: 

https://github.com/msupply-foundation/open-msupply/assets/117138852/cf1eb243-ceac-4dd4-9791-81462b9e43e4

## How to Reproduce 🔨

Steps to reproduce the behaviour:

1. Establish remote authorisation setup
2. Create a new internal order
3. Add items form master list
4. The Requested quantity values are 0
5. Still the `Confirm Sent` button is active (BUG)

## Your environment 🌱

- Open mSupply Version: v1.7-00-RC4
- Legacy mSupply Central Server Version:

- Platform:

- Database type:

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #3527

# 👩🏻‍💻 What does this PR do?
Don't allow users to send Internal Orders if there are only placeholder lines or if there are no lines

## 💌 Any notes for the reviewer?

# 🧪 Testing

- [ ] Create an `Internal Order`
- [ ] Try send without lines
- [ ] See message
- [ ] Add lines from masterlist or just add one line
- [ ] Don't put anything in for requested quantity (should be 0)
- [ ] Try to send
- [ ] See message

# 📃 Documentation

- [ ] **Part of an epic**: documentation will be completed for the feature as a whole
- [x] **No documentation required**: no user facing changes or a bug fix which isn't a change in behaviour
- [ ] **These areas should be updated or checked**: 
  1.
  2.

### CODE REVIEW

File: client/packages/requisitions/src/RequestRequisition/DetailView/Footer/StatusChangeButton.tsx [hunk 1]
```diff
@@ -141,13 +141,17 @@ export const StatusChangeButton = () => {
   const { selectedOption, getConfirmation, lines } = useStatusChangeButton();
   const isDisabled = useRequest.utils.isDisabled();
   const { userHasPermission } = useAuthContext();
-  const t = useTranslation('app');
-  const noLines = lines?.totalCount === 0;
+  const t = useTranslation(['app', 'replenishment']);
+  const cantSend =
+    lines?.totalCount === 0 ||
+    lines?.nodes.some(line => line.requestedQuantity === 0);
```
Comment (lache-melvin) [inline client/packages/requisitions/src/RequestRequisition/DetailView/Footer/StatusChangeButton.tsx:147]:
```suggestion
    lines?.nodes.every(line => line.requestedQuantity === 0);
```

I think it should be fine to confirm with some placeholder lines? Just not only placeholder lines right?

File: client/packages/common/src/intl/locales/en/replenishment.json [hunk 2]
```diff
@@ -83,6 +83,7 @@
   "messages.inbound-shipment-created-on": "Inbound shipment created on {{date}}",
   "messages.internal-order-created-on": "Internal order created on {{date}}",
   "messages.off-hold-confirmation": "This will re-enable status changes",
+  "messages.cant-send-order": "Cannot change the status because there are no lines or because there are only placeholder lines",
```
Comment (lache-melvin) [inline client/packages/common/src/intl/locales/en/replenishment.json:86]: Nit but could it be a more specific message... i.e. Cannot send order/cannot send empty order or something? "Change status" feels vague/not quite right to me from a "send" button? 🤷‍♀️ ~

**Review by lache-melvin — CHANGES_REQUESTED** [review]
Thanks Roxy! I think need to change to allow sending orders where some lines are placeholders, otherwise working great <3

File: client/packages/requisitions/src/RequestRequisition/DetailView/Footer/StatusChangeButton.tsx (hunk 1 above)
Comment (roxy-dao) [reply client/packages/requisitions/src/RequestRequisition/DetailView/Footer/StatusChangeButton.tsx:147]: You're right XD

**Review by lache-melvin — APPROVED** [review]
🔥 🙏

**Pull request merged**